meilies-server/src/main.rs

```diff
@@ -91,8 +91,22 @@ fn send_stream_events(
             if is_accepted {
                 let stream = stream.name.clone();
                 let number = EventNumber(event_number);
-                let event = EventData(value.to_vec());
-                let event = Response::Event { stream, number, event };
+                let buf = match &value {
+                    IVec::Inline(sz, buf) => unsafe {
+                        buf.get_unchecked(..*sz as usize)
+                    },
+                    IVec::Remote { buf } => buf,
+                };
+                let event = RawEvent::new(buf);
+                let event_name = match event.name() {
+                    Ok(name) => name,
+                    Err(err) => {
+                        info!("impossible to parse the event name; {}", err);
+                        continue
```

Comment:
You should completely stop reading things here has this kind of error should never happen.